src/transaction_builder.js

```diff
@@ -517,6 +522,7 @@ TransactionBuilder.fromTransaction = function (transaction, network) {
     txb.__addInputUnsafe(txIn.hash, txIn.index, {
       sequence: txIn.sequence,
       script: txIn.script,
+      value: txIn.value,
```

Comment:
Transaction doesn't currently have value in the input... as it is not encoded in the raw transaction that we output........

Like @dcousens said, we can't fix Multisig order if input is witness... since we need a value.

Reply:
But that's all it needs, and __addInputUnsafe already tolerates `value` being provided in the `opts`. It's so close to making this work, because `witness` tells us the `sigVersion` already. 

I'm not proposing everyone should expect the field to be there for anything else, it at least makes it work for all multisigs, not just P2SH ones (it doesn't work with bare either, I guess?)

Reply:
@afk11 I think why this confused me for so long is I didn't understand that you meant you added `txIn.value` to the `txIn` in the _transaction_ that you _pass_ to `fromTransaction`.

It isn't a natural usage of the API,  but I can understand your motivation given you want everything prepared before you do `.sign`.